### ISSUE TITLE
Cannot create new form

### ISSUE THREAD

**Issue by Jivvy** [nextcloud/forms#189]
Forms version 1.1.0
Nextcloud version 18.0.0

### Steps to reproduce
1. Open a new form
2. Enter name, description, and a question. Long Response was tested
3. Click create

### Expected behaviour
Create the form

### Actual behaviour
[index] Error: Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO `oc_forms_events` (`title`, `description`, `is_anonymous`, `unique`, `access`, `owner`, `created`, `hash`) VALUES(?, ?, ?, ?, ?, ?, ?, ?)' with params ["Title of Form", "Description of form", 0, false, "public", "jivvy", "2020-02-02 02:13:31", "7bJLcI1cVrMFGQSJ"]:

SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'unique' at row 1 at <<closure>>

 0. /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php line 169
    Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException("An exception oc ... 1", Doctrine\DBAL\Dr ... ]})
 1. /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php line 145
    Doctrine\DBAL\DBALException::wrapException(Doctrine\DBAL\Driver\PDOMySql\Driver {}, Doctrine\DBAL\Dr ... ]}, "An exception oc ... 1")
 2. /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php line 1063
    Doctrine\DBAL\DBALException::driverExceptionDuringQuery(Doctrine\DBAL\Driver\PDOMySql\Driver {}, Doctrine\DBAL\Dr ... ]}, "INSERT INTO `oc ... )", {1: "Title of Fo ... "})
 3. /var/www/nextcloud/lib/private/DB/Connection.php line 220
    Doctrine\DBAL\Connection->executeUpdate("INSERT INTO `oc ... )", ["Title of Form" ... "], [2,2,1,2,2,2,2,2])
 4. /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php line 203
    OC\DB\Connection->executeUpdate("INSERT INTO `oc ... )", {dcValue1: "Titl ... "}, {dcValue1: 2,dcV ... 2})
 5. /var/www/nextcloud/lib/private/DB/QueryBuilder/QueryBuilder.php line 215
    Doctrine\DBAL\Query\QueryBuilder->execute()
 6. /var/www/nextcloud/lib/public/AppFramework/Db/QBMapper.php line 126
    OC\DB\QueryBuilder\QueryBuilder->execute()
 7. /var/www/nextcloud/apps/forms/lib/Controller/ApiController.php line 495
    OCP\AppFramework\Db\QBMapper->insert(OCA\Forms\Db\Event {id: null})
 8. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 170
    OCA\Forms\Controller\ApiController->writeForm({id: 0,hash: "", ... "}, {formQuizQuestio ... ]}, [], "create")
 9. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 99
    OC\AppFramework\Http\Dispatcher->executeController(OCA\Forms\Contro ... "}, "writeForm")
10. /var/www/nextcloud/lib/private/AppFramework/App.php line 125
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Forms\Contro ... "}, "writeForm")
11. /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
    OC\AppFramework\App::main("OCA\\Forms\\Controller\\ApiController", "writeForm", OC\AppFramework\ ... {}, {_route: "forms.api.write_form"})
12. <<closure>>
    OC\AppFramework\Routing\RouteActionHandler->__invoke({_route: "forms.api.write_form"})
13. /var/www/nextcloud/lib/private/Route/Router.php line 299
    call_user_func(OC\AppFramework\ ... {}, {_route: "forms.api.write_form"})
14. /var/www/nextcloud/lib/base.php line 1008
    OC\Route\Router->match("/apps/forms/write/form")
15. /var/www/nextcloud/index.php line 38
    OC::handleRequest()

POST /nextcloud/apps/forms/write/form
from 192.168.254.15 by jivvy at 2020-02-02T02:13:31+00:00

## Server configuration detail

**Operating system:** Linux 5.0.0-38-generic #41-Ubuntu SMP Tue Dec 3 00:27:35 UTC 2019 x86_64

**Webserver:** Apache/2.4.38 (Ubuntu) (apache2handler)

**Database:** mysql 5.7.28

**PHP version:** 

7.2.24-0ubuntu0.19.04.2
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, sodium, session, standard, apache2handler, mysqlnd, PDO, xml, apcu, bz2, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, imagick, imap, intl, json, ldap, exif, msgpack, mysqli, pdo_mysql, apc, posix, readline, redis, shmop, SimpleXML, smbclient, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Phar, memcached, libsmbclient, Zend OPcache

**Nextcloud version:** 18.0.0 - 18.0.0.10

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** Nextcloud website

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.4.0
 - activity: 2.11.0
 - admin_audit: 1.8.0
 - apporder: 0.9.0
 - breezedark: 18.0.2
 - bruteforcesettings: 1.5.0
 - cloud_federation_api: 1.1.0
 - cms_pico: 1.0.4
 - comments: 1.8.0
 - dav: 1.14.0
 - deck: 0.8.0
 - extract: 1.2.3
 - federatedfilesharing: 1.8.0
 - files: 1.13.1
 - files_pdfviewer: 1.7.0
 - files_rightclick: 0.15.2
 - files_sharing: 1.10.1
 - files_trashbin: 1.8.0
 - files_versions: 1.11.0
 - files_videoplayer: 1.7.0
 - firstrunwizard: 2.7.0
 - forms: 1.1.0
 - groupfolders: 6.0.1
 - issuetemplate: 0.6.0
 - logreader: 2.3.0
 - lookup_server_connector: 1.6.0
 - metadata: 0.11.1
 - nextcloud_announcements: 1.7.0
 - notes: 3.1.1
 - notifications: 2.6.0
 - oauth2: 1.6.0
 - privacy: 1.2.0
 - provisioning_api: 1.8.0
 - serverinfo: 1.8.0
 - settings: 1.0.0
 - sharebymail: 1.8.0
 - spreed: 8.0.1
 - survey_client: 1.6.0
 - systemtags: 1.8.0
 - tasks: 0.11.3
 - text: 2.0.0
 - theming: 1.9.0
 - twofactor_backupcodes: 1.7.0
 - updatenotification: 1.8.0
 - viewer: 1.2.0
 - workflowengine: 2.0.0
Disabled:
 - encryption
 - federation
 - files_external
 - guests
 - password_policy
 - photos
 - recommendations
 - support
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "localhost",
        "192.168.254.14",
        "47.185.211.240"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "18.0.0.10",
    "overwrite.cli.url": "http:\/\/localhost\/nextcloud",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "htaccess.RewriteBase": "\/nextcloud",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "app_install_overwrite": [
        "sharerenamer",
        "activitylog"
    ],
    "memcache.local": "\\OC\\Memcache\\APCu",
    "maintenance": false
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** false

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

## Client configuration

**Browser:** Brave, but autofill says:  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36

**Operating system:** Windows 10 Pro

## Logs

<details><summary>Web server error log</summary>

```

```
</details>

<details><summary>Nextcloud log</summary>

```
[index] Error: Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO `oc_forms_events` (`title`, `description`, `is_anonymous`, `unique`, `access`, `owner`, `created`, `hash`) VALUES(?, ?, ?, ?, ?, ?, ?, ?)' with params ["Title of Form", "Description of form", 0, false, "public", "jivvy", "2020-02-02 02:13:31", "7bJLcI1cVrMFGQSJ"]:

SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'unique' at row 1 at <<closure>>

 0. /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php line 169
    Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException("An exception oc ... 1", Doctrine\DBAL\Dr ... ]})
 1. /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php line 145
    Doctrine\DBAL\DBALException::wrapException(Doctrine\DBAL\Driver\PDOMySql\Driver {}, Doctrine\DBAL\Dr ... ]}, "An exception oc ... 1")
 2. /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php line 1063
    Doctrine\DBAL\DBALException::driverExceptionDuringQuery(Doctrine\DBAL\Driver\PDOMySql\Driver {}, Doctrine\DBAL\Dr ... ]}, "INSERT INTO `oc ... )", {1: "Title of Fo ... "})
 3. /var/www/nextcloud/lib/private/DB/Connection.php line 220
    Doctrine\DBAL\Connection->executeUpdate("INSERT INTO `oc ... )", ["Title of Form" ... "], [2,2,1,2,2,2,2,2])
 4. /var/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php line 203
    OC\DB\Connection->executeUpdate("INSERT INTO `oc ... )", {dcValue1: "Titl ... "}, {dcValue1: 2,dcV ... 2})
 5. /var/www/nextcloud/lib/private/DB/QueryBuilder/QueryBuilder.php line 215
    Doctrine\DBAL\Query\QueryBuilder->execute()
 6. /var/www/nextcloud/lib/public/AppFramework/Db/QBMapper.php line 126
    OC\DB\QueryBuilder\QueryBuilder->execute()
 7. /var/www/nextcloud/apps/forms/lib/Controller/ApiController.php line 495
    OCP\AppFramework\Db\QBMapper->insert(OCA\Forms\Db\Event {id: null})
 8. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 170
    OCA\Forms\Controller\ApiController->writeForm({id: 0,hash: "", ... "}, {formQuizQuestio ... ]}, [], "create")
 9. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 99
    OC\AppFramework\Http\Dispatcher->executeController(OCA\Forms\Contro ... "}, "writeForm")
10. /var/www/nextcloud/lib/private/AppFramework/App.php line 125
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Forms\Contro ... "}, "writeForm")
11. /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
    OC\AppFramework\App::main("OCA\\Forms\\Controller\\ApiController", "writeForm", OC\AppFramework\ ... {}, {_route: "forms.api.write_form"})
12. <<closure>>
    OC\AppFramework\Routing\RouteActionHandler->__invoke({_route: "forms.api.write_form"})
13. /var/www/nextcloud/lib/private/Route/Router.php line 299
    call_user_func(OC\AppFramework\ ... {}, {_route: "forms.api.write_form"})
14. /var/www/nextcloud/lib/base.php line 1008
    OC\Route\Router->match("/apps/forms/write/form")
15. /var/www/nextcloud/index.php line 38
    OC::handleRequest()

POST /nextcloud/apps/forms/write/form
from 192.168.254.15 by jivvy at 2020-02-02T02:13:31+00:00
```
</details>

<details><summary>Browser log</summary>

ProxyBus.js:65 Proxying an event bus of version 1.1.2 with 0.2.1
t @ ProxyBus.js:65
globals.js:66 Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20
ne @ globals.js:66
issuetemplate:1 [DOM] Found 2 elements with non-unique id #operating-system: (More info: https://goo.gl/9p2vKq) <input id=​"operating-system" type=​"text" class=​"form-control">​ <input id=​"operating-system" type=​"text" class=​"form-control">​
</details>

**Comment by belzebub40k** [comment]
I get the same error when I try to create a Form with Nextcloud 17.0.3 and Forms 1.1.0.

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO `oc_forms_events` (`title`, `description`, `is_anonymous`, `unique`, `access`, `owner`, `created`, `hash`) VALUES(?, ?, ?, ?, ?, ?, ?, ?)' with params ["Test123", "", 0, false, "public", "labi", "2020-02-03 13:22:41", "8SAnGL1cXEq91OOA"]: SQLSTATE[22007]: Invalid datetime format: 1366 Incorrect integer value: '' for column `nextcloud`.`oc_forms_events`.`unique` at row 1

**Comment by nightbluepn** [comment]
https://github.com/nextcloud/forms/issues/176#issuecomment-581546056

**Comment by yuanpeterli** [comment]
I have the same issue with v1.1.0 on nextcloud 18.0.

**Comment by jotoeri** [comment]
Should now be fixed with v1.1.1?

**Comment by slwr** [comment]
With v1.1.1 and Nextcloud 17.0.3 issue does not appear any more.
